hack/crdbversions/main.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v2"
+)
+
+// crdb-versions.yaml structure
+type crdbVersions struct {
+	CrdbVersions []string `yaml:"CrdbVersions"`
+}
+
+type templateData struct {
+	CrdbVersions      []Version
+	LatestCrdbVersion string
+	OperatorVersion   string
+	GeneratedWarning  string
+}
+
+type Version struct {
+	Version, UnderscoreVersion string
+}
+
+func main() {
+	log.SetFlags(0)
+	crdbVersionsFile := flag.String("crdb-versions", "", "YAML file with CRDB versions")
+	operatorVersion := flag.String("operator-version", "", "Current operator version")
+	templateFile := flag.String("template", "", "Template file")
+	flag.Parse()
+
+	if *crdbVersionsFile == "" || *operatorVersion == "" || *templateFile == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	contents, err := ioutil.ReadFile(*crdbVersionsFile)
+	if err != nil {
+		log.Fatalf("Cannot open CRDB version file: %s", err)
+	}
+	var versions crdbVersions
+	if err := yaml.Unmarshal(contents, &versions); err != nil {
+		log.Fatalf("Cannot parse CRDB version file: %s", err)
+	}
+	// Collect all version in order to sort them
+	vs := make([]*semver.Version, len(versions.CrdbVersions))
+	for i, r := range versions.CrdbVersions {
+		v, err := semver.NewVersion(r)
+		if err != nil {
+			log.Fatalf("Cannot convert version `%s`: %s", r, err)
+		}
+		vs[i] = v
+	}
+	sort.Sort(semver.Collection(vs))
+	latestVersion := vs[len(vs)-1]
```

Comment:
How does this work for new releases like 22.2 betas etc? Do we want to be using that as "latest"

Reply:
I'm not sure, tbh. We can add logic to filter out unstable releases from the `latestVersion` logic. Should be trivial to do.

Reply:
Hmm. After [looking into this](https://gist.github.com/rail/c995fdcc230f572b36ce6b277c220cbc) I realized a few issues with using unstable versions:
- We don't publish unstable CRDB releases to RedHat, so we can't include them to the OpenShift metadata
- Unstable DockerHub CRDB repo is `cockroachdb/cockroach-unstable`, so we may need to tweak the logic and the templates in case we want to support betas and pre-releases.

Reply:
If we don't want to support or publish them then lets leave it the way it is for now.

Reply:
I added some logic to handle unstable versions.